### PR TITLE
Feature/rep 101 us data center

### PR DESCRIPTION
**Pull request by dovekie**
@alexanderyoung PTAL

### CODE REVIEW

File: bin/critterget.py [hunk 1]
```diff
@@ -10,22 +10,12 @@
...
-
-# FOR EU PoP USERS ONLY:
-# If you are using Apteligent through its EU data center presence,
-# uncomment the following two lines for baseurl and authbaseurl and
-# comment out the previous two lines for baseurl and authbaseurl
-# baseurl = "https://developers.eu.crittercism.com:443/v1.0/"
-# authbaseurl = "https://developers.eu.crittercism.com/v1.0/"
-
-debug = False
+debug = True
```
Comment (alexanderyoung) [inline bin/critterget.py]: Probably shouldn't merge this in to our codebase.

File: bin/critterget.py [hunk 2]
```diff
@@ -781,6 +770,34 @@ def getCrashCounts(appId,appName):
     return
 
 
+def setDataRegion(sessionKey):
```
Comment (alexanderyoung) [inline bin/critterget.py]: snake_case https://www.python.org/dev/peps/pep-0008/#function-names

File: bin/critterget.py [hunk 3]
```diff
@@ -781,6 +770,34 @@ def getCrashCounts(appId,appName):
...
+    if (debug) : print u'{} MessageType="ApteligentDebug"  Into setDataRegion'.format(DATETIME_OF_RUN)
+
+    try:
+        entities = entity.getEntities(['properties', 'crittercism','api'], namespace=myapp,
+                                        owner='nobody', sessionKey=sessionKey)
+        if (debug) : print u'{} MessageType="CritterDebug" entities are {}'.format(DATETIME_OF_RUN, entities)
+        use_eu = int(entities['dataCenterRegion'].value)
+        if (debug) : print u'{} MessageType="CritterDebug" use_eu is {}'.format(DATETIME_OF_RUN, use_eu)
+    except Exception, e:
+        print u'{} MessageType="ApteligentDebug" Could not get a data region because {} Defaulting to US'.format(DATETIME_OF_RUN, e)
```
Comment (alexanderyoung) [inline bin/critterget.py]: Some of these lines are really long.  Please use an 80 chr ruler.  https://www.python.org/dev/peps/pep-0008/#maximum-line-length

File: bin/critterget.py [hunk 4]
```diff
@@ -801,13 +818,17 @@ def getCredentials(sessionKey):
 
         # return first set of credentials
         if (debug) : print "Entities is ", entities
-        if entities is not None:
+        if entities:
             for i, c in entities.items():
```
Comment (alexanderyoung) [inline bin/critterget.py]: Why are we referencing i here?  Also, what's up with the letters i and c?
Why are we redefining auth multiple times?
Do we not know what the key name is for the parent?
Assuming we want the first item, and we don't know the key name, something like this seems reasonable, but not sure of the use case:
if entities:
  auth = entities.items()[0][1].get('clear_password')

If we know the key we want, it should be much easier though.

Comment (dovekie) [inline bin/critterget.py]: This is old code; I can refactor it but I'm not sure it's worth the time at this point.

File: bin/critterget.py (hunk 2 above)
Comment (dovekie) [inline bin/critterget.py]: Changing the style for only one function seems like it would be confusing.

Comment (alexanderyoung) [inline bin/critterget.py]: This module has a mix of both, let's use PEP8.  Feel free to refactor the other functions in a separate change.

File: bin/critterget.py (hunk 4 above)
Comment (alexanderyoung) [inline bin/critterget.py]: You're working in this loop though, and it's pretty confusing.  Let's fix it please.

File: bin/critterget.py [hunk 5]
```diff
@@ -781,6 +770,35 @@ def getCrashCounts(appId,appName):
     return
 
 
+def setDataRegion(sessionKey):
+    """
+    :param sessionKey: A string generated by Splunk and used to access its internal API
```
Comment (alexanderyoung) [inline bin/critterget.py]: Line too long

File: bin/critterget.py [hunk 6]
```diff
@@ -781,6 +770,35 @@ def getCrashCounts(appId,appName):
...
+    :return: None
+    """
+    use_eu = None
+
+    if (debug) : print u'{} MessageType="ApteligentDebug"  Into setDataRegion'.format(DATETIME_OF_RUN)
```
Comment (alexanderyoung) [inline bin/critterget.py]: Line too long

File: bin/critterget.py [hunk 7]
```diff
@@ -781,6 +770,35 @@ def getCrashCounts(appId,appName):
...
+
+    try:
+        entities = entity.getEntities(['properties', 'crittercism','api'], namespace=myapp,
+                                        owner='nobody', sessionKey=sessionKey)
+        if (debug) : print u'{} MessageType="CritterDebug" entities are {}'.format(DATETIME_OF_RUN, entities)
```
Comment (alexanderyoung) [inline bin/critterget.py]: Line too long

File: bin/critterget.py [hunk 8]
```diff
@@ -781,6 +770,35 @@ def getCrashCounts(appId,appName):
...
+    """
+    :param sessionKey: A string generated by Splunk and used to access its internal API
+    :return: None
+    """
+    use_eu = None
+
+    if (debug) : print u'{} MessageType="ApteligentDebug"  Into setDataRegion'.format(DATETIME_OF_RUN)
+
+    try:
+        entities = entity.getEntities(['properties', 'crittercism','api'], namespace=myapp,
```
Comment (alexanderyoung) [inline bin/critterget.py]: Line too long

**Comment by alexanderyoung** [comment]
There are a bunch of lines way in excess of 80, please resolve them.  I highlighted a few.

**Comment by alexanderyoung** [comment]
LGTM

**Pull request merged**